workflow-core/src/commonMain/kotlin/com/squareup/workflow1/BaseRenderContext.kt

```diff
@@ -207,6 +278,32 @@ public interface BaseRenderContext<out PropsT, StateT, in OutputT> {
     }
   }
 
+  @Deprecated(
+    "Always provide a debugging name",
+    ReplaceWith("eventHandler(\"TODO: debugging name\", onFailedCast, update)")
+  )
+  public fun <E1, E2, E3> eventHandler(
+    update: WorkflowAction<@UnsafeVariance PropsT, StateT, @UnsafeVariance OutputT>.Updater.(
+      E1,
+      E2,
+      E3
+    ) -> Unit
+  ): (E1, E2, E3) -> Unit = eventHandler("eventHandler", update)
+
+  @Suppress("DEPRECATION")
+  public fun <E1, E2, E3> eventHandler(
+    name: String,
+    update: WorkflowAction<@UnsafeVariance PropsT, StateT, @UnsafeVariance OutputT>.Updater.(
+      E1,
+      E2,
+      E3
+    ) -> Unit
+  ): (E1, E2, E3) -> Unit = eventHandler({ name }, update)
+
+  @Deprecated(
+    "Requiring a function to provide a debugging name was absurd, we're sorry.",
```

Comment:
Here (and elsewhere with this pattern) you mean the fact that we made `name` a function? It reads confusing, because we now are requiring you to provide a debugging name, but not via a function.

Reply:
```kotlin
  @Deprecated(
    "Use a static string.",
```